source/src/microapp/cs_MicroappStorage.cpp

```diff
@@ -47,331 +53,335 @@ NRF_FSTORAGE_DEF(nrf_fstorage_t nrf_microapp_storage) =
 };
 
 MicroappStorage::MicroappStorage() { 
+}
 
-	_prevMessage.protocol = 0;
-	_prevMessage.app_id = 0;
-	_prevMessage.payload = 0;
-	_prevMessage.repeat = 0;
+cs_ret_code_t MicroappStorage::init() {
+	_buffer = new uint8_t[MICROAPP_STORAGE_BUF_SIZE];
+	if (_buffer == nullptr) {
+		LOGw("Could not allocate buffer");
+		return ERR_NO_SPACE;
+	}
 
-	_enabled = false;
-	_debug = true;
+	uint32_t nrfCode;
+	nrfCode = nrf_fstorage_init(&nrf_microapp_storage, &nrf_fstorage_sd, NULL);
+	switch (nrfCode) {
+		case NRF_SUCCESS:
+			LOGMicroappInfo("Sucessfully initialized from 0x%08X to 0x%08X", nrf_microapp_storage.start_addr, nrf_microapp_storage.end_addr);
+			break;
+		default:
+			LOGw("NRF error code %u", nrfCode);
+			return ERR_NOT_INITIALIZED;
+			break;
+	}
+
+//	uint8_t testData[10] = {0,1,2,3,4,5,6,7,8,9};
+//	uint8_t* testDataPtr = testData + 3;
+//	uint8_t testDataSize = 4;
+//	nrfCode = nrf_fstorage_write(&nrf_microapp_storage, g_FLASH_MICROAPP_BASE, testDataPtr, testDataSize, NULL);
+//	LOGi("nrf_fstorage_write data=%u size=%u nrfCode=%u", testDataPtr, testDataSize, nrfCode);
+//
+//	testDataPtr = testData + 4;
+//	testDataSize = 3;
+//	nrfCode = nrf_fstorage_write(&nrf_microapp_storage, g_FLASH_MICROAPP_BASE, testDataPtr, testDataSize, NULL);
+//	LOGi("nrf_fstorage_write data=%u size=%u nrfCode=%u", testDataPtr, testDataSize, nrfCode);
+//
+//	testDataPtr = testData + 4;
+//	testDataSize = 4;
+//	nrfCode = nrf_fstorage_write(&nrf_microapp_storage, g_FLASH_MICROAPP_BASE, testDataPtr, testDataSize, NULL);
+//	LOGi("nrf_fstorage_write data=%u size=%u nrfCode=%u", testDataPtr, testDataSize, nrfCode);
+
+	return ERR_SUCCESS;
 }
 
-uint16_t MicroappStorage::init() {
-	_buffer = new uint8_t[MICROAPP_CHUNK_SIZE];
-
-	uint32_t err_code;
-	err_code = nrf_fstorage_init(&nrf_microapp_storage, &nrf_fstorage_sd, NULL);
-	switch(err_code) {
-	case NRF_SUCCESS:
-		LOGi("Sucessfully initialized from 0x%08X to 0x%08X", nrf_microapp_storage.start_addr, nrf_microapp_storage.end_addr);
-		break;
-	default:
-		LOGw("Error code %i", err_code);
-		break;
+cs_ret_code_t MicroappStorage::erase(uint8_t appIndex) {
+	if (_writing) {
+		return ERR_BUSY;
 	}
 
-	return err_code;
-}
+	if (isErased(appIndex)) {
+		return ERR_SUCCESS;
+	}
 
-uint16_t MicroappStorage::erasePages() {
-	uint32_t err_code;
-	err_code = nrf_fstorage_erase(&nrf_microapp_storage, nrf_microapp_storage.start_addr, g_FLASH_MICROAPP_PAGES, NULL);
-	return err_code;
+	uint32_t flashAddress = nrf_microapp_storage.start_addr + appIndex * MICROAPP_MAX_SIZE;
+	LOGMicroappInfo("erase addr=0x%08X size=%u", flashAddress, MICROAPP_MAX_SIZE / 0x1000);
+	uint32_t nrfCode = nrf_fstorage_erase(&nrf_microapp_storage, flashAddress, MICROAPP_MAX_SIZE / 0x1000, NULL);
```

Comment:
- please use `nullptr` instead of NULL for type safety. 
- define magic constant `0x1000` somewhere?

Reply:
Using NULL as per documentation of `nrf_fstorage_erase()`.
Defined flash page size in microapp packets for now. Moving this to config or so was on the TODO list.

Reply:
I'm surprised I haven't added preferred use of `nullptr` to the STYLE.md guide yet. `NULL` is a preprocessor symbol usually defines something like `(void*)0)`, but it is implementation defined. There are subtle differences between `NULL` and `nullptr` due to freedom of implementation. Hence `nullptr` is preferred or possibly an explicit alternative if `nullptr` doesn't lead to intended behaviour.

As `nrf_fstorage_erase` requires a pointer type for that argument it is safe to use `nullptr` rather than `NULL`, result is the same.

Reply:
It is stated in https://github.com/crownstone/bluenet/blob/master/docs/DEVELOPER_GUIDE.md

Reply:
Oh no, neither. :-D

Reply:
changed to nullptr